Upgrading from 3.18.2 to 3.18.3 Results in OutOfMemoryError when using @QuarkusTest with Quarkus Junit 5

### Describe the bug

When upgrading from Quarkus `3.18.2` to `3.18.3` our `@QuarkusTest`s start to throw `OutOfMemoryError`. 

It seems the issue is isolated in `quarkus-junit5` dependency or one of its transitives, when I upgrade the rest of dependencies and leave that, things work fine.

Perhaps something to do with this change?  But I'm not sure...
https://github.com/quarkusio/quarkus/commit/820360362c2ae76fc1e97f541ff85f157a363907

### Expected behavior

Tests pass without `OutOfMemoryError`

### Actual behavior

Tests annotated with `@QuarkusTest` hang and eventually run out of memory with different errors each time.
Example errors:

```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: java.lang.OutOfMemoryError: Java heap space

java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: java.lang.OutOfMemoryError: Java heap space

	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:611)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:695)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: java.lang.OutOfMemoryError: Java heap space

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:354)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:271)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:195)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:578)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:628)
	... 1 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: java.lang.OutOfMemoryError: Java heap space

	at io.quarkus.builder.Execution.run(Execution.java:124)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:350)
	... 6 more
Caused by: java.lang.OutOfMemoryError: Java heap space
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

OSX and Linux

### Output of `java -version`

17.0

### Quarkus version or git rev

3.18.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.12.1

### Additional information

I've also tried version `3.19.0` and the issue has NOT been resolved.

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrderer.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrderer.java
@@ -58,11 +58,11 @@ import io.quarkus.test.junit.main.QuarkusMainTest;
  */
 public class QuarkusTestProfileAwareClassOrderer implements ClassOrderer {
 
-    protected static final String DEFAULT_ORDER_PREFIX_NON_QUARKUS_TEST = "10_";
     protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST = "20_";
     protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_MATCHING_RES = "30_";
     protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE = "40_";
     protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_RESTRICTED_RES = "45_";
+    protected static final String DEFAULT_ORDER_PREFIX_NON_QUARKUS_TEST = "60_";
 
     static final String CFGKEY_ORDER_PREFIX_QUARKUS_TEST = "junit.quarkus.orderer.prefix.quarkus-test";
 

--- a/test-framework/junit5/src/test/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrdererTest.java
+++ b/test-framework/junit5/src/test/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrdererTest.java
@@ -98,8 +98,6 @@ class QuarkusTestProfileAwareClassOrdererTest {
         new QuarkusTestProfileAwareClassOrderer().orderClasses(contextMock);
 
         assertThat(input).containsExactly(
-                nonQuarkusTest1Desc,
-                nonQuarkusTest2Desc,
                 quarkusTest1Desc,
                 quarkusTest1aDesc,
                 quarkusTest2Desc,
@@ -114,7 +112,9 @@ class QuarkusTestProfileAwareClassOrdererTest {
                 quarkusTestWithRestrictedResourceDesc,
                 quarkusTestWithRestrictedResourceDesc2,
                 quarkusTestWithMetaResourceDesc,
-                quarkusTestWithMetaResourceDesc2);
+                quarkusTestWithMetaResourceDesc2,
+                nonQuarkusTest1Desc,
+                nonQuarkusTest2Desc);
     }
 
     @Test
@@ -124,9 +124,9 @@ class QuarkusTestProfileAwareClassOrdererTest {
         List<ClassDescriptor> input = Arrays.asList(quarkusTestDesc, nonQuarkusTestDesc);
         doReturn(input).when(contextMock).getClassDescriptors();
 
-        new QuarkusTestProfileAwareClassOrderer("20_", "30_", "40_", "45_", "60_", Optional.empty()).orderClasses(contextMock);
+        new QuarkusTestProfileAwareClassOrderer("20_", "30_", "40_", "45_", "01_", Optional.empty()).orderClasses(contextMock);
 
-        assertThat(input).containsExactly(quarkusTestDesc, nonQuarkusTestDesc);
+        assertThat(input).containsExactly(nonQuarkusTestDesc, quarkusTestDesc);
     }
 
     @Test

